Epsilon.Host.Frontend/src/components/TermDateFilter.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+	<button class="refresh-button" @click="resetDates">↻</button>
+	<div ref="dateRangeMenu" class="date-range-menu">
+		<button @click="toggleMenu">Date range</button>
+		<div class="menu-content" :class="{ 'menu-visible': menuVisible }">
+			<div class="date-range-filter">
+				<div class="date-input">
+					<label for="startDate">Start date:</label>
+					<input
+						id="startDate"
+						v-model="startDate"
+						type="date"
+						@input="filterItems" />
+				</div>
+				<div class="date-input">
+					<label for="endDate">End date:</label>
+					<input
+						id="endDate"
+						v-model="endDate"
+						type="date"
+						@input="filterItems" />
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script lang="ts" setup>
+import { ref, onMounted, defineProps } from "vue"
+import { useStore } from "vuex"
+const store = useStore()
+
+const props = defineProps<{
+	items: Array<{ start_at: string; end_at: string }> | null
+	modelValue: { name: string }
+	placeholder?: string
+	limit: number
+}>()
+
+const startDate = ref<string | null>(null)
+const endDate = ref<string | null>(null)
+const menuVisible = ref(false)
+const originalItems = ref<Array<{ start_at: string; end_at: string }> | null>(
+	null
+)
+const hasSetInitialItems = ref(false)
+const dateRangeMenuRef = ref<HTMLElement | null>(null)
+
+const setInitialItems = (): void => {
+	if (props.items !== null) {
+		originalItems.value = props.items
+	}
+}
+
+const filterItems = (): void => {
```

Comment:
If I read this correctly. This function selects terms depending on two dates? 
And then the option is given to the user to select an term?

We are not looking for this. 
This date selection will need to go around the term selection. 
Essentially u want to manipulate the `currentTerm` start and end date in the vuex store

Reply:
Yes the function I wrote is to select two dates, and then in the SearchBox.vue component only terms within those dates will be shown. Then with the reset button the filtered dates are removed again and all terms are shown. But how I understand it now, is that you want the user to edit/update the dates of the current selected term is that correct?

Reply:
Correct